ipatests/test_integration/test_installation.py

```diff
@@ -479,3 +483,43 @@ def test_reserved_ip_as_forwarder(self):
         exp_str = ("Invalid IP Address 0.0.0.0: cannot use IANA reserved "
                    "IP address 0.0.0.0")
         assert exp_str in cmd.stdout_text
+
+
+class TestKRAinstallAfterCertRenew(IntegrationTest):
+    """ Test KRA installtion after ca agent cert renewal
+
+    KRA installtion was failing after ca-agent cert gets renewed.
+    This test check if the syptoms no longer exists.
+
+    related ticket: https://pagure.io/freeipa/issue/7288
+    """
+
+    def test_KRA_install_after_cert_renew(self):
+
+        tasks.install_master(self.master)
+
+        # get ca-agent cert contents
+        p12_content = self.master.get_file_contents(paths.DOGTAG_ADMIN_P12)
+
+        # load cert
+        cert = x509.load_der_x509_certificate(p12_content,
```

Comment:
You can't stick pkcs12 data in this method, it requires DER-formatted certificate. This is the code I was referring to before:
https://github.com/freeipa/freeipa/blob/1ef0fe8bb824282c2f48417efda3a60e7c1bf580/ipaserver/install/certs.py#L106-L115